### ISSUE TITLE
Error when applying Completely Safe Rockets to ranged weapon

### ISSUE THREAD

**Issue by p3lim** [Stanzilla/WoWUIBugs#508]
```
62x SharedXML/Scroll/ScrollBoxListView.lua:628: attempt to perform arithmetic on a nil value
[string "@SharedXML/Scroll/ScrollBoxListView.lua"]:628: in function <SharedXML/Scroll/ScrollBoxListView.lua:587>
[string "=(tail call)"]: ?
[string "@SharedXML/Scroll/ScrollBoxListView.lua"]:831: in function `ValidateDataRange'
[string "@SharedXML/Scroll/ScrollBox.lua"]:708: in function `Update'
[string "@SharedXML/Scroll/ScrollBox.lua"]:258: in function `SetScrollPercentageInternal'
[string "@SharedXML/Scroll/ScrollBox.lua"]:250: in function `SetScrollPercentage'
[string "@SharedXML/Scroll/ScrollBox.lua"]:242: in function `ScrollToEnd'
[string "@Blizzard_Professions/Blizzard_ProfessionsCraftingOutputLog.lua"]:369: in function `FinalizePendingResultData'
[string "@Blizzard_Professions/Blizzard_ProfessionsCraftingOutputLog.lua"]:317: in function <...rofessions/Blizzard_ProfessionsCraftingOutputLog.lua:313>

Locals:
self = <table> {
 EnumerateFrames = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:157
 GetFrameExtent = <function> defined @SharedXML/Scroll/ScrollController.lua:39
 SetAcquireLocked = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:335
 TranslateElementDataToUnderlyingData = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:241
 ForEachFrame = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:141
 SetFrameExtent = <function> defined @SharedXML/Scroll/ScrollController.lua:44
 FindElementDataIndex = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:195
 SetHorizontal = <function> defined @SharedXML/Scroll/ScrollController.lua:31
 ForEachElementData = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:183
 scrollBox = Frame {
 }
 GetElementExtentCalculator = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:796
 PrepareScrollToElementDataByPredicate = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:249
 CreateTemplateExtentFromElementData = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:543
 UnregisterEvents = <function> defined @SharedXML/CallbackRegistry.lua:200
 GetExtent = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:151
 SetElementExtentCalculator = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:789
 GetPanExtent = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:163
 IsScrollToDataIndexSafe = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:245
 IsVirtualized = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:571
 Layout = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:124
 GetElementIndent = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:110
 SetElementResetter = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:496
 templateInfoDirty = true
 SetPanExtent = <function> defined @SharedXML/Scroll/ScrollBoxView.lua:51
 HasEqualTemplateInfoExtents = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:727
 SetVirtualized = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:500
 SetElementIndentCalculator = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:106
 GetLayoutFunction = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:114
 SetElementInitializer = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:448
 SetScrollBox = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:135
 ReinitializeFrames = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:535
 RemoveDataProvider = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:283
 IsDataIndexWithinRange = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:946
 dataIndexEnd = 1
 FindElementDataByPredicate = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:199
 GetDataIndexEnd = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:933
 SetInvalidationReason = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:913
 OnDataProviderSort = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:318
 GetSpacing = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:53
 RegisterCallback = <function> defined @SharedXML/CallbackRegistry.lua:110
 GetFrames = <function> defined @SharedXML/Scroll/ScrollBoxView.lua:90
 UnregisterCallback = <function> defined @SharedXML/CallbackRegistry.lua:185
 Release = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:413
 RegisterCallbackWithHandle = <function> defined @SharedXML/CallbackRegistry.lua:154
 initializerList = <table> {
 }
 FindByPredicate = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:207
 GetExtentUnti
```

**Comment by rodneycheung** [comment]
the same thing happens to me every time when I apply rune to the weapon.

```6x SharedXML/Scroll/ScrollBoxListView.lua:628: attempt to perform arithmetic on a nil value
[string "@SharedXML/Scroll/ScrollBoxListView.lua"]:628: in function <SharedXML/Scroll/ScrollBoxListView.lua:587>
[string "=(tail call)"]: ?
[string "@SharedXML/Scroll/ScrollBoxListView.lua"]:831: in function `ValidateDataRange'
[string "@SharedXML/Scroll/ScrollBox.lua"]:708: in function `Update'
[string "@SharedXML/Scroll/ScrollBox.lua"]:258: in function `SetScrollPercentageInternal'
[string "@SharedXML/Scroll/ScrollBox.lua"]:250: in function `SetScrollPercentage'
[string "@SharedXML/Scroll/ScrollBox.lua"]:242: in function `ScrollToEnd'
[string "@Blizzard_Professions/Blizzard_ProfessionsCraftingOutputLog.lua"]:369: in function `FinalizePendingResultData'
[string "@Blizzard_Professions/Blizzard_ProfessionsCraftingOutputLog.lua"]:317: in function <...rofessions/Blizzard_ProfessionsCraftingOutputLog.lua:313>

Locals:
self = <table> {
 EnumerateFrames = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:157
 GetFrameExtent = <function> defined @SharedXML/Scroll/ScrollController.lua:39
 SetAcquireLocked = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:335
 TranslateElementDataToUnderlyingData = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:241
 ForEachFrame = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:141
 SetFrameExtent = <function> defined @SharedXML/Scroll/ScrollController.lua:44
 FindElementDataIndex = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:195
 SetHorizontal = <function> defined @SharedXML/Scroll/ScrollController.lua:31
 ForEachElementData = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:183
 scrollBox = Frame {
 }
 GetElementExtentCalculator = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:796
 PrepareScrollToElementDataByPredicate = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:249
 CreateTemplateExtentFromElementData = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:543
 UnregisterEvents = <function> defined @SharedXML/CallbackRegistry.lua:200
 GetExtent = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:151
 SetElementExtentCalculator = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:789
 GetPanExtent = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:163
 IsScrollToDataIndexSafe = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:245
 IsVirtualized = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:571
 Layout = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:124
 GetElementIndent = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:110
 SetElementResetter = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:496
 templateInfoDirty = true
 SetPanExtent = <function> defined @SharedXML/Scroll/ScrollBoxView.lua:51
 HasEqualTemplateInfoExtents = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:727
 SetVirtualized = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:500
 SetElementIndentCalculator = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:106
 GetLayoutFunction = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:114
 SetElementInitializer = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:448
 SetScrollBox = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:135
 ReinitializeFrames = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:535
 RemoveDataProvider = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:283
 IsDataIndexWithinRange = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:946
 dataIndexEnd = 1
 FindElementDataByPredicate = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:199
 GetDataIndexEnd = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:933
 SetInvalidationReason = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:913
 OnDataProviderSort = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:318
 GetSpacing = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:53
 RegisterCallback = <function> defined @SharedXML/CallbackRegistry.lua:110
 GetFrames = <function> defined @SharedXML/Scroll/ScrollBoxView.lua:90
 UnregisterCallback = <function> defined @SharedXML/CallbackRegistry.lua:185
 Release = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:413
 RegisterCallbackWithHandle = <function> defined @SharedXML/CallbackRegistry.lua:154
 initializerList = <table> {
 }
 FindByPredicate = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:207
 GetExtentUnti```

**Comment by rodneycheung** [comment]
It seems that the error has been fixed since 10.2.5
